zerver/lib/streams.py

```diff
@@ -1097,20 +1103,25 @@ def can_administer_accessible_channel(channel: Stream, user_profile: UserProfile
 
 
 @dataclass
-class StreamsCategorizedByPermissions:
+class BaseStreamsCategorizedByPermissions:
     authorized_streams: list[Stream]
     unauthorized_streams: list[Stream]
+
+
+@dataclass
+class StreamsCategorizedByPermissionsForAddingSubscribers(BaseStreamsCategorizedByPermissions):
```

Comment:
I think you can keep a single class with name `StreamsCategorizedByPermissions`, and have it just have a third field `accessible_permission_denied_streams: list[Stream]` that is just the empty list for normal `filter_stream_authorization` (since there were no permission checks beyond being able to access it), and equals `streams_to_which_user_cannot_add_subscribers` for `filter_stream_authorization_for_adding_subscriptions`.

Reply:
That can be done, but wouldn't it make it things a bit more hard to read? Not sure if it will be evidently clear why `accessible_permission_denied_streams` is empty. Don't think it would be hard to manage the two classes.

Reply:
Not relevant anymore after the last commit changes.